sensor/common/scannerdefinitions/http_handler_test.go

```diff
@@ -0,0 +1,91 @@
+package scannerdefinitions
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type responseWriterMock struct {
+	bytes.Buffer
+	statusCode int
+	headers    http.Header
+}
+
+func NewMockResponseWriter() *responseWriterMock {
+	return &responseWriterMock{
+		headers: make(http.Header),
+	}
+}
+
+func (m *responseWriterMock) Header() http.Header {
+	return m.headers
+}
+
+func (m *responseWriterMock) WriteHeader(statusCode int) {
+	m.statusCode = statusCode
+}
+
+// transportMockFunc is a transport mock that call itself to implement http.Transport's RoundTrip.
+type transportMockFunc func(req *http.Request) (*http.Response, error)
+
+func (f transportMockFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req)
+}
+
+func Test_scannerDefinitionsHandler_ServeHTTP(t *testing.T) {
+	type args struct {
+		writer  *responseWriterMock
+		request *http.Request
+	}
+	tests := []struct {
+		name         string
+		args         args
+		responseBody string
+		statusCode   int
+	}{
+		{
+			name:         "when central replies 200 with content then writer matches",
+			statusCode:   200,
+			responseBody: "the foobar body.",
+			args: args{
+				writer: NewMockResponseWriter(),
+			},
+		},
+		{
+			name:         "when central replies 304 then writer matches",
```

Comment:
when would Central respond with 304?

Reply:
When the bundle modified time is newer than the one requested, as per `If-Modified-Since`.

Reply:
Yeah, why not.